crates/wasmer-provider/src/pipes.rs

```diff
@@ -0,0 +1,102 @@
+use std::{
+    convert::TryFrom,
+    fmt::Debug,
+    io::{self, Read, Seek, Write},
+};
+
+use anyhow::Error;
+use serde::{Deserialize, Serialize};
+use tokio::fs::File;
+use wasmer_wasi::{VirtualFile, WasiFsError};
+
+/// For piping stdio. Stores all output / input in a byte-vector.
+pub struct FilePipe {
+    file: std::fs::File,
+}
+
+impl Debug for FilePipe {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("FilePipe")
+            .field("file", &self.file)
+            .finish()
+    }
+}
+
+impl Serialize for FilePipe {
+    fn serialize<S>(&self, _serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        unimplemented!()
```

Comment:
Isn't this the same as just not including the implementation? Why implement `Serialize` then call `unimplemented!()`?